core/src/main/java/com/google/instrumentation/trace/TraceExporter.java

```diff
@@ -54,6 +71,352 @@
   public abstract void unregisterServiceHandler(String name);
 
   /**
+   * Returns the {@code InProcessDebuggingHandler} that can be used to get useful debugging
+   * information like (active spans, latency based sampled spans, error based sampled spans).
+   *
+   * @return the {@code InProcessDebuggingHandler} or {@code null} if in-process debugging is not
+   *     supported.
+   */
+  @Nullable
+  public abstract InProcessDebuggingHandler getInProcessDebuggingHandler();
+
+  /**
+   * This class allows users to access in-process debugging information like (getting access to all
+   * active spans, support latency based sampled spans and error based sampled spans).
+   *
+   * <p>Only spans with the option {@link Span.Options#RECORD_EVENTS} are recorded.
+   *
+   * <p>Access to active spans is available for all the spans independent on the {@code Span} name.
+   *
+   * <p>Support for latency based sampled spans and error based sampled spans MUST be configured
+   * using {@link #collectSamplesForSpanNames(List)}.
+   */
+  public abstract static class InProcessDebuggingHandler {
+
+    InProcessDebuggingHandler() {}
+
+    /**
+     * Returns the summary of all available in-process debugging data like number of active spans,
+     * number of sampled spans in the latency based samples or error based samples.
+     *
+     * @return the summary of all available in-process debugging data.
+     */
+    public abstract Summary getSummary();
+
+    /**
+     * Returns a list of active spans that are matching the {@code filter}.
+     *
+     * @param filter used to filter the returned spans.
+     * @return a list of active spans that are matching the {@code filter}.
+     */
+    public abstract List<SpanData> getActiveSpans(ActiveSpansFilter filter);
```

Comment:
The `List` return type could imply that the spans are ordered.  I think it would be clearer to either explain the order or return a `Collection<SpanData>`.  The same applies to all of the uses of `List` in this file.